packages/astro/src/runtime/server/render/util.ts

```diff
@@ -22,7 +25,9 @@ const toIdent = (k: string) =>
 	});
 
 export const toAttributeString = (value: any, shouldEscape = true) =>
-	shouldEscape ? String(value).replace(/&/g, '&#38;').replace(/"/g, '&#34;') : value;
+	shouldEscape
+		? String(value).replace(ampersandRegex, '&#38;').replace(doubleQuoteRegex, '&#34;')
+		: value;
```

Comment:
hoisting the regex prevents the regex instance from creating everytime. When testing I didn't quite find a big perf improvement though, but I feel like it might still be worth having.

I think the most significant cost is with `String()` but I haven't figured a refactor for it yet from the consumer side.